src/clients/c++/image_client.cc

```diff
@@ -173,44 +216,57 @@ void
 Postprocess(
     const std::map<std::string, std::unique_ptr<nic::InferContext::Result>>&
         results,
-    const std::vector<std::string>& filenames, const size_t batch_size)
+    const std::vector<std::string>& filenames, const size_t batch_size,
+    const float* shm_addr, size_t offset, const size_t& byte_size, bool use_shm)
 {
-  if (results.size() != 1) {
+  if ((results.size() != 1) && (!use_shm)) {
```

Comment:
It is an option to read directly from the shared memory where the outputs were written, but the GetRawAtCursor, etc. should also work just as well. So I would just have a comment saying something like "Can read outputs directly from shared memory but can also use the client APIs", and then use the client APIs.

Reply:
While using shared memory for output, why is it necessary to populate that output as a part of the results in the response?

Reply:
GetRawAtCursor should just maintain a pointer into the shared memory region and use the data directly from there. We want the client APIs to work for both "raw" and shared-memory outputs.